invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import { withCancel } from "react-invenio-forms";
+import { http } from "../../api/config";
+import { Grid, Message, Container } from "semantic-ui-react";
+import FeaturedCommunity from "./FeaturedCommunity";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/invenio_communities/i18next";
+
+export default class FeaturedCommunities extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: { hits: [] },
+      error: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetch && this.cancellableFetch.cancel();
+  }
+
+  fetchData = async () => {
+    this.cancellableFetch = withCancel(http.get("/api/communities/featured"));
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits });
+    } catch (error) {
+      this.setState({ error: error });
+      console.error(error);
+    }
+  };
+
+  displayError() {
+    const { error } = this.state;
+    return (
+      <Container>
+        <Message negative>
+          <Message.Header>{i18next.t("An error occurred.")}</Message.Header>
+          <p>{error?.response?.data.message}</p>
```

Comment:
Only works with errors from the api that return the error in the format `error.response.data.message`. 

I suggest that you make this function independent of axios. Maybe store only the error message in the component state and here only display it. 

```javascript
// Fetch files and store error message
try {}
catch (error) {
  let errorMessage = error.message;
  if (error.response) {
    errorMessage = error.response.data;
  }
  this.setState({
    error: errorMessage
  })
}
```

Then display only the message:
```javascript
  const { error } = this.state;
  return (
    ...
   <p>{error}</p>
  )

```

Reply:
I see, I will add it, thanks!